### PR TITLE
[Fiber] Reorganize files for DOM renderer to make overlap between fiber/stack clearer

### PR DESCRIPTION
**Pull request by sebmarkbage**
I split the injection so that we can continue using the event/property system from fiber without pulling in everything.

I also reorganize files in my best guess for which files we can reuse instead of forking.

### CODE REVIEW

File: src/renderers/dom/shared/ReactDOMInjection.js [hunk 1]
```diff
@@ -6,7 +6,7 @@
  * LICENSE file in the root directory of this source tree. An additional grant
  * of patent rights can be found in the PATENTS file in the same directory.
  *
- * @providesModule ReactDefaultInjection
+ * @providesModule ReactDOMInjection
```
Comment (sebmarkbage) [inline src/renderers/dom/shared/ReactDOMInjection.js]: @iamdustan This file might be of interest to you. It is the remainder of the injection system. At least in the first version of Fiber, but we can probably get rid of these completely at some point.

File: src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js [hunk 2]
```diff
@@ -626,8 +626,10 @@ describe('ReactDOMComponent', () => {
     }
 
     beforeEach(() => {
-      var ReactDefaultInjection = require('ReactDefaultInjection');
-      ReactDefaultInjection.inject();
+      var ReactDOMInjection = require('ReactDOMInjection');
+      ReactDOMInjection.inject();
+      var ReactDOMStackInjection = require('ReactDOMInjection');
```
Comment (iamdustan) [inline src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js]: should the require statement here by `ReactDOMStackInjection`?

**Comment by iamdustan** [comment]
:thumbsup: I like the change of `Default` -> `DOM` making the renderer aspect much more explicit.

It appears to me like the `React[Renderer]StackInjection` is a stepping stone while the reconciler code lives in the renderers rather than core to the future Fiber renderer API.

**Comment by nhunzaker** [comment]
Just a spectator but I think confusion around the injection was one of the biggest things that threw me off when first contributing. This is fantastic.

File: src/renderers/dom/shared/ReactDOMInjection.js (hunk 1 above)
Comment (sebmarkbage) [inline src/renderers/dom/shared/ReactDOMInjection.js]: Ah. Thanks. Updated. But I guess I don't technically need it to pass this test. Will revisit when we try to make this test work with Fiber.

File: src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js (hunk 2 above)
Comment (sebmarkbage) [inline src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js]: Ah. Thanks. Updated. But I guess I don't technically need it to pass this test. Will revisit when we try to make this test work with Fiber.

**Comment by sebmarkbage** [comment]
I verified that I was able to build Fiber with the `/shared` dependencies without the files in `/stack` (with a few tweaks for `ReactUpdates`).

**Comment by gaearon** [comment]
Accepting to unblock, will properly review tomorrow morning.

**Comment by spicyj** [comment]
lgtm

**Comment by gaearon** [comment]
Read through, all looks good.

**Pull request merged**